### ISSUE TITLE
Make promote-menhir failing, cannot build parser

### ISSUE THREAD

**Issue by mmcqd** [ocaml/ocaml#11591]
Hello, I'm trying to re-generate the parser. Eventually I want to modify the parser, but I'm running into errors trying to build it without even modifying it. The full set of commands I've run is as follows (from the root of the repo)
```
./configure
make
make install
make promote-menhir
```
The first three succeed without issue. The final step is failing with the following error
```
menhir --lalr --explain --dump --require-aliases --strict -lg 1 -la 1 --unused-token COMMENT --unused-token DOCSTRING --unused-token EOL --unused-token GREATERRBRACKET  --infer --ocamlc "./boot/ocamlrun ./boot/ocamlc -nostdlib -I ./boot -use-prims runtime/primitives " --fixed-exception --table --strategy simplified  parsing/parser.mly
File "parsing/parser.mly", line 748, characters 17-27:
Error: Unbound module Location
make: *** [promote-menhir] Error 1
```
It seems that menhir is unaware of any of the modules in `parsing/` directory, since deleting all references to `Location` in `parser.mly` results in errors about `Asttypes` being unbound. Am I doing something obviously wrong?

**Comment by dra27** [comment]
This looks to be an unintended consequence of 7de4d2b6 (cc @shindere).

A temporary workaround (I haven't looked fully) may be to add this to `Makefile.menhir`:

```make
COMPFLAGS=-strict-sequence -principal -absname \
          -w +a-4-9-40-41-42-44-45-48 \
          -warn-error +a \
          -bin-annot -safe-string -strict-formats $(INCLUDES)
```

**Comment by mmcqd** [comment]
Thanks for the fast response @dra27. U̶n̶f̶o̶r̶t̶u̶n̶a̶t̶e̶l̶y̶ ̶y̶o̶u̶r̶ ̶s̶u̶g̶g̶e̶s̶t̶i̶o̶n̶ ̶d̶o̶e̶s̶n̶'̶t̶ ̶f̶i̶x̶ ̶t̶h̶e̶ ̶p̶r̶o̶b̶l̶e̶m̶.̶ ̶I̶'̶l̶l̶ ̶s̶e̶e̶ ̶w̶h̶a̶t̶ ̶I̶ ̶c̶a̶n̶ ̶f̶i̶g̶u̶r̶e̶ ̶o̶u̶t̶ ̶f̶r̶o̶m̶ ̶l̶o̶o̶k̶i̶n̶g̶ ̶a̶t̶ ̶t̶h̶e̶ ̶c̶o̶m̶m̶i̶t̶ ̶y̶o̶u̶ ̶p̶o̶i̶n̶t̶e̶d̶ ̶t̶o̶.̶

EDIT: Actually your suggestion did work 😄 . I just had to define `COMPFLAGS` before it got used haha. Thank you!

Replacing `$(COMPFLAGS)` with `$(OC_COMMON_CFLAGS) $(INCLUDES)` in `Makefile.menhir` also worked, and I'm assuming that's the solution that's in line with @shindere's changes. I can open a PR fixing this.

**Comment by shindere** [comment]
Oops, sorry, I didn't see the suggestion because it was an EDIT and GitHub does not send out e-mails for that) and thus I opened #11592 which exactly implements the suggested change. Thanks for the report and sorry for having messed things up by overlooking `Makefile.menhir` .

**Comment by dra27** [comment]
Fixed in #11592

**Comment by shindere** [comment]
David Allsopp (2022/10/02 04:29 -0700):
> This looks to be an unintended consequence of 7de4d2b6 (cc @shindere).

Indeed. Sorry for having overlooked `Makefile.menhir` and messed things
up.

I just submitted PR #11592 which I believe fixes this issue.
